src/__swaps__/screens/Swap/hooks/useNativeAssetForChain.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react';
+
+import { ExtendedAnimatedAssetWithColors } from '@/__swaps__/types/assets';
+import { ChainId } from '@/__swaps__/types/chains';
+
+import { SharedValue, runOnJS, runOnUI, useAnimatedReaction, useDerivedValue, useSharedValue } from 'react-native-reanimated';
+import { ParsedAddressAsset } from '@/entities';
+import { ethereumUtils } from '@/utils';
+
+export const useNativeAssetForChain = ({ inputAsset }: { inputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null> }) => {
+  const chainId = useDerivedValue(() => inputAsset.value?.chainId ?? ChainId.mainnet);
+  const nativeAsset = useSharedValue<ParsedAddressAsset | undefined>(
+    ethereumUtils.getNetworkNativeAsset(ethereumUtils.getNetworkFromChainId(chainId.value))
+  );
+
+  const getNativeAssetForNetwork = useCallback(
+    (chainId: ChainId) => {
+      const updateNativeAsset = (asset: ParsedAddressAsset | undefined) => {
+        'worklet';
+        nativeAsset.value = asset;
+      };
+
+      const network = ethereumUtils.getNetworkFromChainId(chainId);
+      const asset = ethereumUtils.getNetworkNativeAsset(network);
+      runOnUI(updateNativeAsset)(asset);
```

Comment:
why does this need to be run on UI? can't the sharedvalue value be updated from any context?

Reply:
this is a good question.. cc @christianbaroni

do we need UI wrappers to update SharedValues?